decoder/label_schema_test.go

```diff
@@ -0,0 +1,256 @@
+			schema: &schema.BodySchema{
+				Blocks: map[string]*schema.BlockSchema{
+					"resource": {
+						Labels: []*schema.LabelSchema{
+							{
+								Name:        "type",
+								IsDepKey:    true,
+								Completable: true,
+							},
+						},
+						DependentBody: map[schema.SchemaKey]*schema.BodySchema{
+							schema.NewSchemaKey(schema.DependencyKeys{
+								Labels: []schema.LabelDependent{
+									{Index: 0, Value: "aws_instance"},
+								},
+							}): {
+								Attributes: map[string]*schema.AttributeSchema{
+									"ami": {
+										Expr:       schema.LiteralTypeOnly(cty.String),
+										IsRequired: true,
+									},
+									"instance_type": {Expr: schema.LiteralTypeOnly(cty.String)},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: lang.CompleteCandidates([]lang.Candidate{
+				{
+					Label: "aws_instance",
+					TextEdit: lang.TextEdit{
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start: hcl.Pos{
+								Line:   1,
+								Column: 11,
+								Byte:   10,
+							},
+							End: hcl.Pos{
+								Line:   1,
+								Column: 11,
+								Byte:   10,
+							},
+						},
+						NewText: "aws_instance",
+						Snippet: "aws_instance",
+					},
+					Kind: lang.LabelCandidateKind,
+				},
+			}),
+		},
+		{
+			name: "prefill required attributes",
+			cfg:  "resource \"\" {\n}",
+			position: hcl.Pos{
+				Line:   1,
+				Column: 11,
+				Byte:   10,
+			},
+			prefillRequiredAttributes: true,
+			schema: &schema.BodySchema{
+				Blocks: map[string]*schema.BlockSchema{
+					"resource": {
+						Labels: []*schema.LabelSchema{
+							{
+								Name:        "type",
+								IsDepKey:    true,
+								Completable: true,
+							},
+							{
+								Name:        "type",
+								IsDepKey:    true,
+								Completable: true,
+							},
+						},
+						DependentBody: map[schema.SchemaKey]*schema.BodySchema{
+							schema.NewSchemaKey(schema.DependencyKeys{
+								Labels: []schema.LabelDependent{
+									{Index: 0, Value: "aws_instance"},
+								},
+							}): {
+								Attributes: map[string]*schema.AttributeSchema{
+									"ami": {
+										Expr:       schema.LiteralTypeOnly(cty.String),
+										IsRequired: true,
+									},
+									"instance_type": {Expr: schema.LiteralTypeOnly(cty.String)},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: lang.CompleteCandidates([]lang.Candidate{
+				{
+					Label: "aws_instance",
+					TextEdit: lang.TextEdit{
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start: hcl.Pos{
+								Line:   1,
+								Column: 11,
+								Byte:   10,
+							},
+							End: hcl.Pos{
+								Line:   1,
+								Column: 14,
+								Byte:   13,
+							},
+						},
+						NewText: "aws_instance\" \"${2:type}\" {\n\tami = \"${3:value}\"\n\t${0}",
+						Snippet: "aws_instance\" \"${2:type}\" {\n\tami = \"${3:value}\"\n\t${0}",
+					},
+					Kind: lang.LabelCandidateKind,
+				},
+			}),
+		},
```

Comment:
Do you mind adding one more test which exercises pre-filling of required attributes within a nested block, or modifying the existing one with `spec` block?

Reply:
Done!